packages/core/src/utils/Utils.ts

```diff
@@ -514,6 +515,35 @@ export class Utils {
       || !!new Error().stack!.split('\n').find(line => line.match(/\w\.ts:\d/)); // as a last resort, try to find a TS file in the stack trace
   }
 
+  private static async getJestConfig(): Promise<Dictionary> {
+    if (await pathExists(process.cwd() + '/jest.config.json')) {
+      return require(process.cwd() + '/jest.config.json');
+    }
+    return {};
+  }
+
+  private static jestConfigUsesTsJest(jestConfig: Dictionary): boolean {
+    return jestConfig.preset === 'ts-jest'
+      || (Object.values(jestConfig.transform ?? {}).some(transformer => transformer === 'ts-jest'));
+  }
+
+  /**
+   * Tries to detect `ts-jest` runtime.
+   */
+  static async detectTsJest(): Promise<boolean> {
+    if (process.argv.length > 1 && process.argv[1].includes('jest-worker')) {
+      const packageConfig = await ConfigurationLoader.getPackageConfig();
+      const jestConfig = await Utils.getJestConfig();
+      if (
+        (packageConfig.jest && Utils.jestConfigUsesTsJest(packageConfig.jest))
+        || Utils.jestConfigUsesTsJest(jestConfig)
+      ) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
this could be re-written to reduce complexity:

```suggestion
    if (process.argv.length <= 1 || !process.argv[1].includes('jest-worker')) {
        return false;
    }

    const packageConfig = await ConfigurationLoader.getPackageConfig();
    const jestConfig = await Utils.getJestConfig();

    return (packageConfig.jest && Utils.jestConfigUsesTsJest(packageConfig.jest)) || Utils.jestConfigUsesTsJest(jestConfig);
```

even better `(packageConfig.jest && Utils.jestConfigUsesTsJest(packageConfig.jest))` - the first condition should be handled in `jestConfigUsesTsJest` method